rpcadaptors.go

```diff
@@ -324,3 +328,405 @@ func (b *rpcSyncMgr) ProcessTransaction(tx *dcrutil.Tx, allowOrphans bool,
 	return b.blockMgr.ProcessTransaction(tx, allowOrphans,
 		rateLimit, allowHighFees, tag)
 }
+
+// rpcUtxoEntry represents a utxo entry for use with the RPC server and
+// implements the rpcserver.UtxoEntry interface.
+type rpcUtxoEntry struct {
+	utxoEntry *blockchain.UtxoEntry
+}
+
+// Ensure rpcUtxoEntry implements the rpcserver.UtxoEntry interface.
+var _ rpcserver.UtxoEntry = &rpcUtxoEntry{}
```

Comment:
I've never seen it done this way. I believe the idiomatic way to check that something satisfies an interface is so: `var _  rpcserver.UtxoEntry= (*rpcUtxoEntry)(nil)`

Reply:
Ya, good point - I followed what was already being done in that file for the other adaptors, which works, but I believe the way its currently being done will allocate memory for an empty struct whereas the way you outlined would not. I'll update those across this file